### ISSUE TITLE
May I use your UNET model as a example in a public presentation...

### ISSUE THREAD

**Issue by aurotripathy** [jocicmarko/ultrasound-nerve-segmentation#66]
Its at the embedded vision summit...and I will provide attribution to your github code.
I find it to be the cleanest and simplest expression of the model. 
I would has expected the conv2d parameter to be 'valid' instead of 'same' as I suspect the original authors implemented it in Caffe. But that's OK.

Kindly indicate if you will support this.

**Comment by jocicmarko** [comment]
Of course, it's fine, I released this code with MIT license, so feel free to use it.

Good luck with your presentation!

**Comment by aurotripathy** [comment]
Thank you! I'm using a variation that (I think) is closer to the paper (added bottom-of-U conv w/1024 feature maps, conv5_B). I also lined up the left convs with the right convs. You quick review would be appreciated (though not necessary)

<code>
def get_orig_unet():

    inputs = Input((img_rows, img_cols, 1))                                                                                  
    conv1_L = Conv2D(64, (3, 3), activation='relu', padding='same')(inputs)
    conv1_L = Conv2D(64, (3, 3), activation='relu', padding='same')(conv1_L)
    pool1 = MaxPooling2D(pool_size=(2, 2))(conv1_L)

    conv2_L = Conv2D(128, (3, 3), activation='relu', padding='same')(pool1)
    conv2_L = Conv2D(128, (3, 3), activation='relu', padding='same')(conv2_L)
    pool2 = MaxPooling2D(pool_size=(2, 2))(conv2_L)  

    conv3_L = Conv2D(256, (3, 3), activation='relu', padding='same')(pool2)
    conv3_L = Conv2D(256, (3, 3), activation='relu', padding='same')(conv3_L) 
    pool3 = MaxPooling2D(pool_size=(2, 2))(conv3_L)                                            

    conv4_L = Conv2D(512, (3, 3), activation='relu', padding='same')(pool3)
    conv4_L = Conv2D(512, (3, 3), activation='relu', padding='same')(conv4_L)  
    pool4 = MaxPooling2D(pool_size=(2, 2))(conv4_L)
  
    conv5_B = Conv2D(1024, (3, 3), activation='relu', padding='same')(pool4)  
    conv5_B = Conv2D(512, (3,3), activation='relu', padding='same')(conv5_B)  

    up1 = concatenate([Conv2DTranspose(512, (2, 2), strides=(2, 2), padding='same')(conv5_B), conv4_L], axis=3)                                                                          
    conv3_R = Conv2D(512, (3, 3), activation='relu', padding='same')(up1)
    conv3_R = Conv2D(512, (3, 3), activation='relu', padding='same')(conv3_R)

    up2 = concatenate([Conv2DTranspose(256, (2, 2), strides=(2, 2), padding='same')(conv3_R), conv3_L], axis=3) # 24, 24                                                                                     
    conv2_R = Conv2D(256, (3, 3), activation='relu', padding='same')(up2)
    conv2_R = Conv2D(256, (3, 3), activation='relu', padding='same')(conv2_R)

    up3 = concatenate([Conv2DTranspose(128, (2, 2), strides=(2, 2), padding='same')(conv2_R), conv2_L], axis=3) # 48, 48                                                                                     
    conv1_R = Conv2D(128, (3, 3), activation='relu', padding='same')(up3)
    conv1_R = Conv2D(128, (3, 3), activation='relu', padding='same')(conv1_R)

    up4 = concatenate([Conv2DTranspose(64, (2, 2), strides=(2, 2), padding='same')(conv1_R), conv1_L], axis=3)                                                                                     
    conv_out = Conv2D(64, (3, 3), activation='relu', padding='same')(up4)
    conv_out = Conv2D(64, (3, 3), activation='relu', padding='same')(conv_out)

    conv_out_final = Conv2D(1, (1, 1), activation='sigmoid')(conv_out)

    model = Model(inputs=[inputs], outputs=[conv_out_final])

    model.compile(optimizer=Adam(lr=1e-5), loss=dice_coef_loss, metrics=[dice_coef])

    return model


</code>
